### ISSUE TITLE
Make Production Ready and Contribute back to Spring Session?

### ISSUE THREAD

**Issue by rwinch** [ldoguin/spring-session-couchbase#1]
Hey it has been a while since we last talked! I stumbled upon this project and was wondering if you were still interested in contributing this back to Spring Session?

**Comment by ldoguin** [comment]
Yes absolutely. Got side tracked by lots of other things. From what I remember there was some work to do on the event part. Creating a phantom doc holding the TTL so that we could actually sent the document to be deleted in the appropriate event.
And I am sure there are other things to do :)

**Comment by rwinch** [comment]
@ldoguin Thanks for the fast response! 

One thing to note is that we can always start with an implementation that does not support events. The events are not required by the API and not everyone wants or needs the events anyways.

**Comment by ldoguin** [comment]
@rwinch Actually the event part is already splitted and lives under spring-session-couchbase-listener.

**Comment by rwinch** [comment]
@ldoguin Thats great news! Do you think you can start exploring getting this into a branch of Spring Session? I went ahead and created https://github.com/spring-projects/spring-session/issues/363 for tracking this from the Spring Session side

Things we typically look for when getting a new datastore (looks like this is already pretty close):
- [x] Code (obviously)
- [x] Unit tests
- [x] Integration tests (fewer than the unit tests for speed but still ensure things are working)
- [ ] Hello World Sample / Guide for Web Application (i.e. [HttpSession Sample](http://docs.spring.io/spring-session/docs/current/reference/html5/guides/httpsession.html))
- [ ] Sample Tests
- [x] Optional `@EnableCouchbaseHttpSession` for making things easier
- [ ] Optional spring-session-couchbase pom that aggregates the dependencies to make dependency management for users easier
